src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostClaimByIdControllerTest.java

```diff
@@ -536,6 +607,69 @@ public void user_should_claim_task_when_grant_type_challenged(
         );
     }
 
+    @ParameterizedTest
+    @CsvSource(value = {
+        "IA, Asylum",
+        "WA, WaCaseType"
+    })
+    public void user_should_claim_task_when_grant_type_challenged_with_granular_permission(
+        String jurisdiction, String caseType)
+        throws Exception {
+
+        mockServices.mockUserInfo();
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+
+        RoleAssignmentRequest roleAssignmentRequest = RoleAssignmentRequest.builder()
+            .testRolesWithGrantType(TestRolesWithGrantType.CHALLENGED_ACCESS_JUDICIARY_PUBLIC)
+            .roleAssignmentAttribute(
+                RoleAssignmentAttribute.builder()
+                    .jurisdiction(jurisdiction)
+                    .caseType(caseType)
+                    .caseId("claimCaseId1")
+                    .build()
+            )
+            .build();
+
+        createRoleAssignment(roleAssignments, roleAssignmentRequest);
+
+        RoleAssignmentResource accessControlResponse = new RoleAssignmentResource(roleAssignments);
+
+        TaskRoleResource taskRoleResource = new TaskRoleResource(
+            TestRolesWithGrantType.CHALLENGED_ACCESS_JUDICIARY_PUBLIC.getRoleName(),
+            false, false, true, false, false, false,
```

Comment:
Just curious and general question, are we still supporting "Refer" permission? There was a ticket recently to remove this from our DMN

Reply:
> As it has been removed from DMN, in our data model it will always be false. So it is a candidate to be removed and has to be done in a separate ticket if needed.